### PR TITLE
feat(coverage): expose raw v8 coverage

### LINKED ISSUE #955: [Feature] Expose Rawer V8 coverage information
see also this issue from puppeteer: https://github.com/puppeteer/puppeteer/issues/2136

I've been trying to get puppeteer-to-instanbul to work with playwright - and I see the puppeteer and playwright implementations are the same - but I'm not getting very good information back - playwright destroys function call information, collapses the ranges and doesn't record call counts - which makes the underlying library v8-to-istanbul (which is used afaik by the node teams and as an experimental feature in jest) not viable.

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/955

### CODE REVIEW

File: src/chromium/crCoverage.ts [hunk 1]
```diff
@@ -82,7 +99,7 @@ class JSCoverage {
     this._resetOnNavigation = resetOnNavigation;
     this._reportAnonymousScripts = reportAnonymousScripts;
     this._enabled = true;
-    this._scriptURLs.clear();
+    this._scriptIds.clear();
     this._scriptSources.clear();
     this._eventListeners = [
       helper.addEventListener(this._client, 'Debugger.scriptParsed', this._onScriptParsed.bind(this)),
```
Comment (lukeapage) [inline src/chromium/crCoverage.ts]: Thanks for this! I've tested a simple hack with raw coverage and it works nicely, apart from:

5 lines down from this we have:
```
this._client.send('Profiler.startPreciseCoverage', {callCount: false, detailed: true}),
```
I can get a really nice coverage report with istanbul if we get callCount info:
```
this._client.send('Profiler.startPreciseCoverage', {callCount: true, detailed: true}),
```

Comment (pavelfeldman) [reply src/chromium/crCoverage.ts]: Sounds good, counting calls now.

File: src/chromium/crCoverage.ts [hunk 2]
```diff
@@ -217,77 +232,27 @@ class CSSCoverage {
...
     helper.removeEventListeners(this._eventListeners);
 
-    // aggregate by styleSheetId
-    const styleSheetIdToCoverage = new Map();
-    for (const entry of ruleTrackingResponse.ruleUsage) {
-      let ranges = styleSheetIdToCoverage.get(entry.styleSheetId);
-      if (!ranges) {
-        ranges = [];
-        styleSheetIdToCoverage.set(entry.styleSheetId, ranges);
+    const entries = new Map<string, CSSCoverageEntry>();
```
Comment (aslushnikov) [inline src/chromium/crCoverage.ts]: These changes are scary: I'm not sure CSS coverage actually yields disjoint ranges (and IIRC we had some problems here).

Can we keep the old codepath for CSS coverage?

File: test/golden-chromium/csscoverage-involved.txt [hunk 3]
```diff
@@ -11,6 +11,6 @@
         "end": 433
       }
     ],
-    "text": "\n@charset \"utf-8\";\n@namespace svg url(http://www.w3.org/2000/svg);\n@font-face {\n  font-family: \"Example Font\";\n  src: url(\"./Dosis-Regular.ttf\");\n}\n\n#fluffy {\n  border: 1px solid black;\n  z-index: 1;\n  /* -webkit-disabled-property: rgb(1, 2, 3) */\n  -lol-cats: \"dogs\" /* non-existing property */\n}\n\n@media (min-width: 1px) {\n  span {\n    -webkit-border-radius: 10px;\n    font-family: \"Example Font\";\n    animation: 1s identifier;\n  }\n}\n"
+    "source": "\n@charset \"utf-8\";\n@namespace svg url(http://www.w3.org/2000/svg);\n@font-face {\n  font-family: \"Example Font\";\n  src: url(\"./Dosis-Regular.ttf\");\n}\n\n#fluffy {\n  border: 1px solid black;\n  z-index: 1;\n  /* -webkit-disabled-property: rgb(1, 2, 3) */\n  -lol-cats: \"dogs\" /* non-existing property */\n}\n\n@media (min-width: 1px) {\n  span {\n    -webkit-border-radius: 10px;\n    font-family: \"Example Font\";\n    animation: 1s identifier;\n  }\n}\n"
```
Comment (aslushnikov) [inline test/golden-chromium/csscoverage-involved.txt]: can we keep "text"?

**Pull request merged**